src/DynamoCore/Core/DynamoMigrator.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Xml.Serialization;
+using Dynamo.Models;
+using ProtoCore.Lang;
+
+namespace Dynamo.Core
+{
+    public struct FileVersion
+    {
+        public readonly int MajorPart;
+        public readonly int MinorPart;
+
+        public FileVersion(int majorPart, int minorPart)
+        {
+            MajorPart = majorPart;
+            MinorPart = minorPart;
+        }
+    }
+
+    /// <summary>
+    /// Base class for all versions of Dynamo classes relating to
+    /// migrating preferences, packages and definitions from older versions
+    /// </summary>
+    internal abstract class DynamoMigratorBase
+    {
+        protected readonly string rootFolder;
+
+        /// <summary>
+        /// Returns the path to the user AppData directory for the current version
+        /// </summary>
+        public string CurrentVersionPath
+        {
+            get
+            {
+                return Path.Combine(rootFolder, String.Format("{0}.{1}", DynamoVersion.MajorPart, DynamoVersion.MinorPart));
+            }
+        }
+
+        public abstract FileVersion DynamoVersion { get; }
+        
+        protected DynamoMigratorBase()
+        {
+        }
+
+        protected DynamoMigratorBase(string rootFolder)
+        {
+            this.rootFolder = rootFolder;
+        }
```

Comment:
The default constructors are to be removed, right? I understand they are used because of `Activator` failure you saw before. The role of any `DynamoMigratorBase` derived class should be to migrate both `%AppData%` and `%ProgramData%` into newer version, are we handling those two here?